### PR TITLE
[HCB Wrapped] Add reimbursement and card grant metrics

### PR DESCRIPTION
**Pull request by YodaLightsabr**
## Summary of the problem
HCB Wrapped needs additional metrics for 2024

## Describe your changes
Adds the following metrics (with maybe more to come):
- Reimbursement count
- Reimbursement amount
- Card grant count
- Card grant amount

### CODE REVIEW

File: app/models/metric/user/card_grant_amount.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#
+#  index_metrics_on_subject  (subject_type,subject_id)
+#
+class Metric
+  module User
+    class CardGrantAmount < Metric
+      include Subject
+
+      def calculate
+        user.card_grants.sum(&:amount_cents)
```
Comment (garyhtou) [inline app/models/metric/user/card_grant_amount.rb]:
```suggestion
        user.card_grants.sum(:amount_cents)
```

when sum takes a symbol, it sum the column using SQL. when it takes a block, it sum in Ruby by calling the block (method).

In our case here, we can just use SQL to sum

File: app/models/metric/user/reimbursement_amount.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#
+#  index_metrics_on_subject  (subject_type,subject_id)
+#
+class Metric
+  module User
+    class ReimbursementAmount < Metric
+      include Subject
+
+      def calculate
+        user.reimbursement_reports.sum(&:amount_cents)
```
Comment (garyhtou) [inline app/models/metric/user/reimbursement_amount.rb]: This can stay as a block (`&:amount_cents`) since we don't have a column for it

**Review by garyhtou — APPROVED** [review]
🪄

File: app/models/metric/user/card_grant_amount.rb (hunk 1 above)
Comment (YodaLightsabr) [reply app/models/metric/user/card_grant_amount.rb]: Oh that's right. Thanks for catching this.

**Pull request merged**